beam_mapping/include/beam_mapping/Utils.h

```diff
@@ -14,12 +15,49 @@ namespace beam_mapping {
 /** @addtogroup mapping
  *  @{ */
 
+using scan_data_type =
+    std::pair<std::vector<Eigen::Matrix4d>, std::vector<PointCloud::Ptr> >;
```

Comment:
I like this
```
struct ScanPose {
Eigen::Matrix4d T_WORLD_LIDAR;
PointCloud::Ptr cloud_in_LIDAR
}
using scan_data_type = std::set<ScanPose>
```
I'll go with it

Reply:
We actually already have a ScanPose struct in beam_slam: https://github.com/BEAMRobotics/beam_slam/blob/main/bs_models/include/bs_models/lidar/scan_pose.h

But it's a large class. Not sure if it makes sense to port that over to here, or keep this more simple. Only issue I see is double naming might cause confusion. Maybe call this ScanAndPose just so that the name is different? Actually I might liek that better because it's more clean what it is

Reply:
The reason for the current choice of data structure follows from the current design of the code. As you mentioned, the current way of storing timestamps, poses, and scans lead me to this design choice: I will create a seperate issue where this is improved on.

Reply:
renamed `scan_data_type` to `pose_and_scan_data_type`. Naming convention follows your style of naming in beam_mapping. I will flag 'fix naming conventions' as an item in a future pull request.